native/cairo_prover/src/lib.rs

```diff
@@ -105,4 +106,34 @@ fn cairo_verify(proof: Vec<u8>, public_input: Vec<u8>) -> bool {
     verify_cairo_proof(&proof, &pub_inputs, &proof_options)
 }
 
-rustler::init!("Elixir.Cairo.CairoProver", [cairo_prove, cairo_verify]);
+#[rustler::nif(schedule = "DirtyCpu")]
+fn cairo_get_compliance_output(public_input: Vec<u8>) -> NifResult<Vec<Vec<u8>>>{
+    let (pub_inputs, _) : (PublicInputs, usize) =
+        bincode::serde::decode_from_slice(&public_input, bincode::config::standard()).unwrap();
+    let output_segments = match pub_inputs.memory_segments.get(&SegmentName::Output) {
+        Some(segment) => segment,
+        None => {
+            eprintln!("Error: 'Output' segment not found in memory_segments");
+            return Ok(vec![]);
+        }
+    };
+
+    let begin_addr :u64 = output_segments.begin_addr.try_into().unwrap();
+    let stop_addr :u64 = output_segments.stop_ptr.try_into().unwrap();
+
+    let mut output_values = Vec::new();
+    for addr in begin_addr..=stop_addr-1 {
```

Comment:
```suggestion
    for addr in begin_addr..stop_addr {
```
It seems identical but more concise?